blob/azureblob/example_test.go

```diff
@@ -42,22 +42,33 @@ func Example() {
 	// Creates a *blob.Bucket backed by Azure Storage Account
 	b, _ := azureblob.OpenBucket(ctx, serviceURL, bucketName, &azureOpts)
 
-	// Obtain the native blob attributes
+	// Get the native blob attributes
+	// See https://godoc.org/github.com/Azure/azure-storage-blob-go/azblob#BlobGetPropertiesResponse
 	var nativeAttrs azblob.BlobGetPropertiesResponse
 	attr, _ := b.Attributes(ctx, key)
 	if attr.As(&nativeAttrs) {
-		// Use the native SDK object
+		// Use the native SDK type
 	}
 
-	// Obtain the Azure SDK type for container which represents a go-cloud bucket
-	var nativeBucket azblob.ContainerURL
+	// Get the native type from a bucket
+	// See https://godoc.org/github.com/Azure/azure-storage-blob-go/azblob
+	var nativeBucket *azblob.ContainerURL
 	if b.As(&nativeBucket) {
-		// Use the native SDK object
+		// Use the native SDK type
 	}
 
 	// Write data to a new blob
+	beforeWrite := func(as func(i interface{}) bool) error {
+		// See https://godoc.org/github.com/Azure/azure-storage-blob-go/azblob#UploadStreamToBlockBlobOptions
+		var nativeWriteOptions *azblob.UploadStreamToBlockBlobOptions
+		if as(&nativeWriteOptions) {
+			// Use the native SDK type
+		}
+		return nil
+	}
 	opts := &blob.WriterOptions{
 		ContentType: "application/json",
+		BeforeWrite: beforeWrite,
 	}
 	w, _ := b.NewWriter(ctx, key, opts)
```

Comment:
Since reading/writing aren't the key points for this example,consider modifying it to use `b.WriteAll` and `b.ReadAll`.

Reply:
Or at least `WriteAll`, you need the `Reader`.

Reply:
Done.